File: solutions/cpp/006_pallindrome.cpp
#include <bits/stdc++.h>
using namespace std;

bool isStringPallindrome(string originalString) {
    string reversedString = "";
    for (int i = originalString.length() - 1; i >= 0; i--) {
        reversedString = reversedString + originalString[i];
    }

    if (reversedString == originalString) {
        return true;
    } else {
        return false;
    }
}

bool isNumberPallindrome(long long originalNumber) {
    long long reversedNumber = 0;
    long long temp = originalNumber;

    while (temp > 0) {
        int digit = temp % 10;
        reversedNumber = (reversedNumber * 10) + digit;
        temp = temp / 10;
    }

    if (reversedNumber == originalNumber) {
        return true;
    } else {
        return false;
    }
    
}

int main() {
    
    long long number;
    cout << "Enter your number: ";
    cin >> number;

    if (isNumberPallindrome(number)) {
        cout << number << " is PALLINDROME!!!";
    } else {
        cout << number << " is NOT PALLINDROME!!!";
    }

    return 0;
}

File: solutions/python/006_pallindrome.py
def isStringPallindrome(STRING):
    if (STRING[::-1] == STRING):
        return True
    else:
        return False
def isNumberPallindrome(NUMBER):
    if (str(NUMBER)[::-1] == str(NUMBER)):
        return True
    else:
        return False

if (isNumberPallindrome(1112)):
    print("YES it is pallindrome!")
else:
    print("NO IT IS NOT A PALLINDROME!")